wasm-rpc/src/json/mod.rs

```diff
@@ -39,6 +39,77 @@ struct TypeAnnotatedValueJson {
     value: serde_json::Value,
 }
 
+/// A representation that optionally pairs type definition with a JSON represented value.
+///
+/// It can only be converted to any of the typed value representations if the type information
+/// is present (or provided externally).
+///
+/// The JSON format is backward compatible with `TypeAnnotatedValueJson`.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct OptionallyTypeAnnotatedValueJson {
+    typ: Option<AnalysedType>,
```

Comment:
let's call this 'type' in the json representation instead of 'typ'

Reply:
I don't want to break the existing API (this new type is "the old" just making the `typ` field optional)